Update reference documentation for Worker accessible DO interfaces (#17411)
cloudflare/cloudflare-docs: src/content/docs/durable-objects/api/alarms.mdx

Notably:

- Each Durable Object instance is able to schedule a single alarm at a time by calling `setAlarm()`.
- Alarms have guaranteed at-least-once execution and are retried automatically when the `alarm()` handler throws.

> - Retries are performed using exponential backoff starting at a two second delay from the first failure with up to six retries allowed.

Reviewer: nit: I don't know what our policy is for numbers, but "2s" or even "2 seconds" seems better than "two seconds".  Same with 6/six later on.

### getAlarm

- <code>getAlarm()</code>: <Type text ='number | null' />

>   - If there is an alarm set, then return the currently set alarm time in number of milliseconds elapsed since the UNIX epoch. Otherwise, return `null`.

Reviewer: ```suggestion
  - If there is an alarm set, then return the currently set alarm time as the number of milliseconds elapsed since the UNIX epoch. Otherwise, return `null`.
```

### setAlarm

- <code> setAlarm(scheduledTimeMs <Type text="number" />)</code>: <Type text="void" />

>   - Set the time for the alarm to run at in number of milliseconds elapsed since the UNIX epoch.

Reviewer: ```suggestion
  - Set the time for the alarm to run.  Specify the time as the number of milliseconds elapsed since the UNIX epoch.
```

### deleteAlarm

- `deleteAlarm()`: <Type text='void' />

  - Unset the alarm if there is a currently set alarm.

  - Calling `deleteAlarm()` inside the `alarm()` handler may prevent retries on a best-effort basis, but is not guaranteed.

## Handler methods

### alarm

- `alarm()`: <Type text='void' />

  - Called by the system when a scheduled alarm time is reached.

  - The `alarm()` handler has guaranteed at-least-once execution and will be retried upon failure using exponential backoff, starting at two second delays for up to six retries. Retries will be performed if the method fails with an uncaught exception.

  - This method can be `async`.

## Example

This example shows how to both set alarms with the `setAlarm(timestamp)` method and handle alarms with the `alarm()` handler within your Durable Object.

- The `alarm()` handler will be called once every time an alarm fires.

> - If an unexpected error terminates the Durable Object, the `alarm()` handler will be re-instantiated on another machine.

Reviewer: "will be" or "may be"?  With SQLite-backed DOs, re-instantiating the DO will likely be on the same metal.

### getAlarm

- `getAlarm()`: <Type text ='number | null' />

  - If there is an alarm set, then return the currently set alarm time in number of milliseconds elapsed since the UNIX epoch. Otherwise, return `null`.

### setAlarm

> - <code>

Reviewer 2: Line breaks were breaking the format. Have adjusted in my own commit.
